--- a/include/linux/mfd/mt6397/core.h
+++ b/include/linux/mfd/mt6397/core.h
@@ -72,10 +72,14 @@
 	u16 int_con[3];
 	u16 int_status[3];
 	u16 chip_id;
+	bool irq_suspended;
+	bool irq_wake_enabled;
 	void *irq_data;
 };
 
 int mt6358_irq_init(struct mt6397_chip *chip);
 int mt6397_irq_init(struct mt6397_chip *chip);
+int mt6323_irq_suspend(struct device *dev);
+int mt6323_irq_resume(struct device *dev);
 
 #endif /* __MFD_MT6397_CORE_H__ */
